client/dashboard/domains/contact-details/contact-form.tsx

```diff
@@ -0,0 +1,342 @@
+						value={ getValue( { item: data } ) }
+						onChange={ ( value ) => onChange( { [ id ]: value } ) }
+					/>
+				);
+			}
+
+			// If the item data is not in the statesList, set the state to the first option
+			if ( ! statesList?.some( ( state ) => state.code === getValue( { item: data } ) ) ) {
+				onChange( { [ id ]: statesList[ 0 ]?.code } );
+			}
+
+			return (
+				<SelectControl
+					__next40pxDefaultSize
+					__nextHasNoMarginBottom
+					label={ hideLabelFromVision ? '' : __( 'State' ) }
+					value={ getValue( { item: data } ) }
+					options={
+						statesList.map( ( state ) => ( {
+							label: state.name,
+							value: state.code,
+						} ) ) ?? []
+					}
+					onChange={ ( value ) => onChange( { [ id ]: value } ) }
+				/>
+			);
+		},
+		[ statesList ]
+	);
+
+	const fields: Field< DomainContactDetails >[] = [
+		{
+			id: 'firstName',
+			label: __( 'First Name' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'lastName',
+			label: __( 'Last Name' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'organization',
+			label: __( 'Organization' ),
+			type: 'text',
+		},
+		{
+			id: 'email',
+			label: __( 'Email' ),
+			type: 'email',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'phone',
+			label: __( 'Phone' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'countryCode',
+			label: __( 'Country' ),
+			type: 'select',
+			elements: formattedCountryList,
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'address1',
+			label: __( 'Address' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'address2',
+			label: __( 'Address Line 2' ),
+			type: 'text',
+		},
+		{
+			id: 'city',
+			label: __( 'City' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'state',
+			label: __( 'State' ),
+			type: 'text',
+			getValue: ( { item }: { item: DomainContactDetails } ) => item.state ?? '',
+			Edit: StateFieldEdit,
+		},
+		{
+			id: 'postalCode',
+			label: __( 'Post Code' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'optOutTransferLock',
+			label: __( 'Opt-out of the 60-day transfer lock' ),
+			type: 'boolean',
+			Edit: ( { field, onChange, data, hideLabelFromVision } ) => {
+				const { id, getValue } = field;
+				return (
+					<CheckboxControl
+						label={
+							hideLabelFromVision
+								? ''
+								: createInterpolateElement(
+										sprintf(
+											/* translators: %s: "what is this?" link */
+											__( 'Opt-out of the 60-day transfer lock <link>%s</link>' ),
+											__( 'what is this?' )
+										),
+										{
+											link: <InlineSupportLink supportContext="60-day-transfer-lock" />,
+										}
+								  )
+						}
+						checked={ getValue( { item: data } ) }
+						onChange={ () => onChange( { [ id ]: ! getValue( { item: data } ) } ) }
+					/>
```

Comment:
```suggestion
					<HStack spacing={ 0 } alignment="start" justify="flex-start">
						<CheckboxControl
							label=""
							checked={ getValue( { item: data } ) }
							onChange={ () => onChange( { [ id ]: ! getValue( { item: data } ) } ) }
						/>
						{ ! hideLabelFromVision && (
							<Text>
								{ createInterpolateElement(
									sprintf(
										/* translators: %s: "what is this?" link */
										__( 'Opt-out of the 60-day transfer lock <link>%s</link>' ),
										__( 'what is this?' )
									),
									{
										link: <InlineSupportLink supportContext="60-day-transfer-lock" />,
									}
								) }
							</Text>
						) }
					</HStack>
```
Since label only support strings, I wonder If we could do something similar, but i'm not sure if this is a good approach. Maybe @fushar has a better idea 😅

Reply:
Good catch, thanks for the suggestion!